src/Symfony/Bundle/SecurityBundle/MagicLink/FirewallAwareMagicLoginLinker.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\MagicLink;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Bundle\SecurityBundle\Security\FirewallMap;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\MagicLink\MagicLoginLinkDetails;
+use Symfony\Component\Security\Http\MagicLink\MagicLoginLinkerInterface;
+
+/**
+ * Decorates the magic login linker for the current firewall.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class FirewallAwareMagicLoginLinker implements MagicLoginLinkerInterface
+{
+    private $firewallMap;
+    private $magicLinkerLocator;
+    private $requestStack;
+
+    public function __construct(FirewallMap $firewallMap, ContainerInterface $magicLinkerLocator, RequestStack $requestStack)
+    {
+        $this->firewallMap = $firewallMap;
+        $this->magicLinkerLocator = $magicLinkerLocator;
+        $this->requestStack = $requestStack;
+    }
+
+    public function createMagicLink(UserInterface $user): MagicLoginLinkDetails
+    {
+        return $this->getMagicLinker()->createMagicLink($user);
+    }
+
+    public function consumeMagicLink(string $uri): UserInterface
+    {
+        return $this->getMagicLinker()->consumeMagicLink($uri);
+    }
+
+    private function getMagicLinker(): MagicLoginLinkerInterface
+    {
+        if (null === $request = $this->requestStack->getCurrentRequest()) {
+            throw new \LogicException('Cannot determine the correct MagicLinker to use: there is no active Request and so, the firewall cannot be determined. Try using the specific magic login linker service.');
+        }
+
+        $firewallName = $this->firewallMap->getFirewallConfig($request)->getName();
+        if (!$this->magicLinkerLocator->has($firewallName)) {
+            throw new \InvalidArgumentException(sprintf('No magic linker found. Did you add a magic_link key under your "%s" firewall?', $firewallName));
```

Comment:
Here we refer to `magic linker`, but the previous error mentions `MagicLinker`. Should we standardize this?